### PR TITLE
Smart charging add amps to consumption curve

### PR DESCRIPTION
**Pull request by OliveGerste**

### CODE REVIEW

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts [hunk 1]
```diff
@@ -5,15 +5,32 @@ import { AppDatePipe } from 'app/shared/formatters/app-date.pipe';
...
 import { ChargingStation, ChargingStationPowers } from 'app/types/ChargingStation';
+import { ConsumptionUnit } from 'app/types/Transaction';
+import { ChargingStations } from 'app/utils/ChargingStations';
 import { Utils } from 'app/utils/Utils';
 import { Chart, ChartColor, ChartData, ChartDataSets, ChartOptions, ChartPoint, ChartTooltipItem } from 'chart.js';
 import * as moment from 'moment';
 
 import { AppDecimalPipe } from '../../../../shared/formatters/app-decimal-pipe';
 
+export interface UnitButtonGroup {
```
Comment (mahdibenromdhane) [inline src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts:16]: This interface is used multiple time, it should be In shared

File: src/app/pages/organization/site-areas/list/consumption-chart/site-area-consumption-chart.component.html [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+      </mat-datetimepicker>
+      <mat-datetimepicker-toggle matSuffix [for]="picker"></mat-datetimepicker-toggle>
+      <input matInput placeholder="{{'site_areas.consumption_date' | translate}}" required
+        (dateChange)="dateFilterChanged($event.value ? $event.value.toDate() : null)" [matDatetimepicker]="picker"
+        [formControl]="dateControl" autocomplete="false">
+      <mat-error *ngIf="dateControl.errors?.matDatepickerParse">{{"general.invalid_date" | translate}}
+      </mat-error>
+    </mat-form-field>
+  </div>
+  <div class="col-auto" [hidden]="siteAreaConsumption?.values?.length === 0">
```
Comment (mahdibenromdhane) [inline src/app/pages/organization/site-areas/list/consumption-chart/site-area-consumption-chart.component.html:14]: You can use the method canDisplayGraph()

File: src/app/pages/organization/site-areas/list/consumption-chart/site-area-consumption-chart.component.ts [hunk 3]
```diff
@@ -13,6 +16,12 @@ import { LocaleService } from '../../../../../services/locale.service';
 import { AppDatePipe } from '../../../../../shared/formatters/app-date.pipe';
 import { AppDecimalPipe } from '../../../../../shared/formatters/app-decimal-pipe';
 
+export interface UnitButtonGroup {
+  key: string;
+  description: string;
+  inactive: boolean;
+}
```
Comment (mahdibenromdhane) [inline src/app/pages/organization/site-areas/list/consumption-chart/site-area-consumption-chart.component.ts:23]: Same as above

File: src/assets/i18n/fr.json [hunk 4]
```diff
@@ -1808,15 +1808,19 @@
...
+      "energy": "Energy Delivered (kW.h)",
+      "reset_zoom": "Reset",
+      "no_consumption": "No consumption data available",
+      "cumulated_amount": "Amount",
+      "limit_plan_watts": "Plan Limit (kW)",
+      "limit_watts": "Max Limit (kW)",
+      "limit_amps": "Max Limit (A)",
+      "load_all_consumptions": "Load all consumption values (without optimization)",
+      "unit_kilowatts": "kW",
+      "unit_amperage":  "Amp"
```
Comment (mahdibenromdhane) [inline src/assets/i18n/fr.json:1823]: It's the fr.json not the en.json

**Comment by mahdibenromdhane** [comment]
On slow internet, the chart take some time to be dislplayed, maybe we can add a spinner.

File: src/assets/i18n/fr.json [hunk 5]
```diff
@@ -1812,15 +1812,20 @@
...
+      "reset_zoom": "Reset",
+      "no_consumption": "No consumption data available",
+      "cumulated_amount": "Amount",
+      "limit_plan": "Plan Limit",
+      "limit": "Max Limit",
+      "limit_watts": "Max Limit (kW)",
+      "limit_amps": "Max Limit (A)",
+      "load_all_consumptions": "Load all consumption values (without optimization)",
+      "unit_kilowatts": "kW",
+      "unit_amperage":  "Amp"
```
Comment (jerome-benoit) [inline src/assets/i18n/fr.json:1828]: Just add the new entries to translate, do not delete the previous ones already correctly translated.

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts [hunk 6]
```diff
@@ -130,6 +144,9 @@ export class ChargingStationSmartChargingLimitPlannerChartComponent implements O
           y: chargingSlot.limitInkW,
         } as number & ChartPoint);
       }
+      if (this.selectedUnit === ConsumptionUnit.AMPERE) {
+        chargingSlotDataSet.data.forEach((chartPoint) => chartPoint.y = chartPoint.y * 1000 / 230 ); // Utils convert method was updated. Needs to be handled by Serge.
```
Comment (jerome-benoit) [inline src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts:148]: It's not that hard to just merge with the master-qa branch and follow the API changes of the method: it's just one argument added to the conversion function.

File: src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts [hunk 7]
```diff
@@ -163,6 +180,9 @@ export class ChargingStationSmartChargingLimitPlannerChartComponent implements O
           y: chargingStationPowers.currentWatt / 1000,
         } as number & ChartPoint);
       }
+      if (this.selectedUnit === ConsumptionUnit.AMPERE) {
+        limitDataSet.data.forEach((chartPoint) => chartPoint.y = chartPoint.y * 1000 / 230 ); // Utils convert method was updated. Needs to be handled by Serge.
```
Comment (jerome-benoit) [inline src/app/pages/charging-stations/charging-limit/charging-profile-limit/charging-station-charging-profile-limit-chart.component.ts:184]: It's not that hard to just merge with the master-qa branch and follow the API change of the method: it's just one argument added to the conversion function.

**Pull request merged**